Cache HDR screen property in WebContent on iOS.

pvollan opened the pull request:
#### f6b4e11f8ba6fc95d85737b8ed6b09547ce2812b
<pre>
Cache HDR screen property in WebContent on iOS.
<a href="https://bugs.webkit.org/show_bug.cgi?id=253724">https://bugs.webkit.org/show_bug.cgi?id=253724</a>

Reviewed by Jer Noble.

This screen property should be sent from the UI process and cached in the WebContent process.

* Source/WebCore/platform/ios/PlatformScreenIOS.mm:
(WebCore::collectScreenProperties):

Canonical link: <a href="https://commits.webkit.org/261535@main">https://commits.webkit.org/261535@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d976fc8a46d6fe8b00ef6334fe6382d0f439686a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/111951 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/21077 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/562 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/3713 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120624 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/116011 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/22427 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/12226 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/3458 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117711 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/16647 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99809 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/104970 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/98599 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/370 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/45634 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/13517 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/387 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/92876 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/14190 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/9799 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/19564 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52389 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15998 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/4378 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

jernoble commented on Source/WebCore/platform/ios/PlatformScreenIOS.mm:
```diff
@@ -82,6 +82,9 @@ bool screenSupportsExtendedColor(Widget*)
 {
 #if USE(MEDIATOOLBOX)
+    if (auto data = screenData(primaryScreenDisplayID()))
+        return data->screenSupportsHighDynamicRange;
```
This should be moved outside the `#if USE` macro block.

pvollan replied:
Done!

Thanks for reviewing!

pvollan committed:
Cache HDR screen property in WebContent on iOS.
https://bugs.webkit.org/show_bug.cgi?id=253724

Reviewed by Jer Noble.

This screen property should be sent from the UI process and cached in the WebContent process.

* Source/WebCore/platform/ios/PlatformScreenIOS.mm:
(WebCore::collectScreenProperties):

Canonical link: https://commits.webkit.org/261535@main
